Maple2.Server.Game/PacketHandlers/SuperChatHandler.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Linq;
+using Maple2.Database.Storage;
+using Maple2.Model.Game;
+using Maple2.PacketLib.Tools;
+using Maple2.Server.Core.Constants;
+using Maple2.Server.Core.PacketHandlers;
+using Maple2.Server.Game.Packets;
+using Maple2.Server.Game.Session;
+
+namespace Maple2.Server.Game.PacketHandlers;
+
+public class SuperChatHandler : PacketHandler<GameSession> {
+    public override RecvOp OpCode => RecvOp.SuperWorldChat;
+
+    private enum Command : byte {
+        Select = 0,
+        Deselect = 1,
+    }
+
+    public override void Handle(GameSession session, IByteReader packet) {
+        var function = packet.Read<Command>();
+        switch (function) {
+            case Command.Select:
+                HandleSelect(session, packet);
+                return;
+            case Command.Deselect:
+                HandleDeselect(session);
+                return;
+        }
+    }
+
+    private static void HandleSelect(GameSession session, IByteReader packet) {
+        int itemId = packet.ReadInt();
+
+        Item? superChatItem = session.Item.Inventory.Find(itemId).First();
+        if (!int.TryParse(superChatItem.Metadata.Function?.Parameters.Split(",").First(), out int superChatId)) {
+            return;
+        }
+
+        session.Player.Value.Character.SuperChatId = superChatId;
```

Comment:
I feel like we should just store the `itemId` which makes it easier to consume the item when it's used.

Reply:
so like save both superChatId and ItemId?

Reply:
you don't need to save `superChatId` because with `itemId` you can get `ItemMetadata` which gives you the same `superChatId` again

Reply:
But we do need to save it. it goes in the FieldAddUser packet. this is so if a user changes fields, their superchat is still enabled/selected

Reply:
Oh I see, it's fine to save both then if we need it